Skip already built check if archive_site is empty

l2dy opened the pull request:
Hack for https://trac.macports.org/ticket/56110. e.g. Use `./mpbb list-subports --archive-site= git` in mpbot.

Hint for myself: exclude reasons are printed to stderr, so this is the only hack needed.

raimue commented on mpbb-list-subports:
```diff
@@ -48,7 +48,7 @@ print-subports() {
         # shellcheck disable=SC2154
         archive_path=$("${tclsh}" "${thisdir}/tools/archive-path.tcl" "${port}")
-        if [[ -f "${archive_path}" ]]; then
+        if [[ -f "${archive_path}" && -n "${archive_site}" ]]; then
```
Couldn't we also skip `archive-path.tcl` if `archive_site` is empty?

l2dy replied:
Yes, but I'd like to make a rather small change. Feel free to push a better version instead.

raimue replied:
I pushed a new commit to the pull request branch. That seemed easier than trying to explain what I meant, which was only to move the conditional.

raimue committed:
mpbb-list-subports: Allow empty --archive-site

Passing an empty string to --archive-site disables the checks on the
remote mirror.

See: https://trac.macports.org/ticket/56110

l2dy commented:
Merged as 868f68814515c2af4dc4c4921b489b7ded410e99.
